### DISCUSSION TITLE
[Bug] overrides.yaml doesn't use yaml syntax

### DISCUSSION THREAD

**Discussion by im-Kitsch** [facebookresearch/hydra#2114] [General]
Hi, 

firstly thanks a lot for the effort.

I tried to read overries.yaml, but it doesn't use yaml sytax. 

For simplest case, 
python example.py a=1 b=2 c=3

then there will be .hydra/overrides.yaml like this:
```
- a=1
- b=2
- c=3
```
But it could not the loaded as yaml file.  While the other files config.yaml and hydra.yaml could be rightly loaded as yaml files. 

Is it a bug or property?

Best.

Should it be

**Comment by Jasha10** [comment]
It is valid yaml; you can load it like this:
```python
>>> import yaml
>>> data = """
... - a=1
... - b=2
... - c=3
... """
>>> yaml.safe_load(data)
['a=1', 'b=2', 'c=3']
```
Or like this:
```python
>>> from omegaconf import OmegaConf
>>> OmegaConf.create(data)
['a=1', 'b=2', 'c=3']
>>> OmegaConf.load("overrides.yaml")
['a=1', 'b=2', 'c=3']
```

**Reply by im-Kitsch** [reply]
Hi,

thanks for the fast replay. 

yes, I know it is possible to be read so. But I wanna to know if it could be read as Dict class like normal yaml files but not like list.

Since it will be quite helpful for management of sweeping over different experiments.

Best.

**Reply by Jasha10** [reply]
Hydra does not expose an API for this.
You could parse the data yourself:
```python
>>> import yaml
>>> data = """
... - a=1
... - b=2
... - c=3
... """
>>> pydata = yaml.safe_load(data)
>>> pydata = dict(s.split("=") for s in pydata)
>>> pydata
{'a': '1', 'b': '2', 'c': '3'}
```
Hydra parses overrides according to the [override grammar](https://hydra.cc/docs/advanced/override_grammar/basic/#grammar).
Internally, Hydra uses an antlr parser to do the work:
```python
>>> from hydra.core.override_parser.overrides_parser import OverridesParser
>>> parser = OverridesParser.create()
>>> overrides = parser.parse_overrides(["a=1", "b=2", "c=3"])
>>> print(overrides)
...
```
This is not part of Hydra's public API and might change in the future.

**Reply by im-Kitsch** [reply]
thanks a lot for the reply, it's quite helpful!
